devimint/src/util.rs

```diff
@@ -107,31 +100,60 @@ pub struct ProcessHandleInner {
     child: Option<Child>,
 }
 
+impl ProcessHandleInner {
+    async fn terminate(&mut self) -> anyhow::Result<()> {
+        if let Some(child) = self.child.as_mut() {
+            debug!(
+                target: LOG_DEVIMINT,
+                name=%self.name,
+                signal="SIGTERM",
+                "sending signal to terminate child process"
+            );
+
+            send_sigterm(child);
+
+            if (fedimint_core::runtime::timeout(Duration::from_secs(2), child.wait()).await)
+                .is_err()
```

Comment:
given fedimintd sleeps for 10s on shutdown, this will always send SIGKILL?

Reply:
I fixed that. Now it shutdowns in <1s.